rts/motoko-rts/src/stabilization/layout/stable_bigint.rs

```diff
@@ -0,0 +1,202 @@
+use crate::bigint::mp_calloc;
+use crate::memory::Memory;
+use crate::stabilization::deserialization::stable_memory_access::StableMemoryAccess;
+use crate::stabilization::layout::write_padding_u64;
+use crate::stabilization::serialization::stable_memory_stream::{
+    ScanStream, StableMemoryStream, WriteStream,
+};
+use crate::stabilization::serialization::SerializationContext;
+use crate::tommath_bindings::mp_digit;
+use crate::types::{size_of, BigInt, Bytes, Value, TAG_BIGINT};
+
+use super::{round_to_u64, Serializer, StableToSpace, StableValue, StaticScanner};
+
+// Tom's math library, as configured for Motoko RTS, encodes a big numbers as an array of 32-bit
+// elements, where each element stores 28 bits of the number while its highest 4 bits are zero.
+// Similar to the little endian encoding, the element array starts with the least significant 28 bits,
+// with each subsequent element covering the next higher 28 bits.
+
+// Tom's math library functions `mp_to_sbin` and `mp_from_sbin` imply temporary object allocations
+// which could be problematic when memory is short.
+// Therefore, a custom serialization to the defined portable stable bignum format is implemented.
+
+#[repr(C)]
+pub struct StableBigInt {
+    is_negative: bool, // Sign bit.
+    number_of_bits: Bits, // Number of used bits in the payload.
+
+                       // Dynamically sized payload of the big integer stored in little endian format.
+                       // Zero padding to align to the next `u64`.
+}
+
+const USED_BITS_PER_ELEMENT: u32 = 28;
```

Comment:
Is this 28 on both 64- and 32-bit?

Reply:
The encoding is still 32-bit in Tom's math library. The number consists of `mp_digit` that is `u32` even on 64-bit. I therefore also compute the elements of the stable bigints as `i32` and pad it to 64-bit.

Reply:
**Update**: I now use the 64-bit version of Tom's math library for EOP. This uses 60bits per 64-bit word, thus being  more space-efficient.